sys/include/net/gnrc/ipv6/nib/ft.h

```diff
@@ -0,0 +1,149 @@
+ *                  (is handed over to a reactive routing protocol if one exist
+ *                  on the interface found and no route is found)
+ * @param[out] fte  The resulting forwarding table entry.
+ *
+ * @return  0, on success.
+ * @return  -ENETUNREACH, if no route was found.
+ */
+int gnrc_ipv6_nib_ft_get(const ipv6_addr_t *dst, gnrc_pktsnip_t *ctx,
+                         gnrc_ipv6_nib_ft_t *fte);
+
+/**
+ * @brief   Adds a new route to the forwarding table
+ *
+ * @param[in] dst       The destination to the route. May be NULL or `::` for
+ *                      default route.
+ * @param[in] dst_len   The prefix length of @p dst in bits. May be 0 for
+ *                      default route.
+ * @param[in] next_hop  The next hop to @p dst/@p dst_len. May be NULL, if
+ *                      @p dst/@p dst_len is no the default route.
+ * @param[in] iface     The interface to @p next_hop. May not be 0.
+ *
+ * @return  0, on success.
+ * @return  -EINVAL, if a parameter was of invalid value.
+ * @return  -ENOMEM, if there was no space left in forwarding table.
+ */
+int gnrc_ipv6_nib_ft_add(const ipv6_addr_t *dst, unsigned dst_len,
+                         const ipv6_addr_t *next_hop, unsigned iface);
+
+/**
+ * @brief   Deletes a route from forwarding table.
+ *
+ * @param[in] dst       The destination of the route. May be NULL or `::` for
+ *                      default route.
+ * @param[in] dst_len   The prefix length of @p dst in bits. May be 0 for
+ *                      default route.
+ */
+void gnrc_ipv6_nib_ft_del(const ipv6_addr_t *dst, unsigned dst_len);
+
+/**
+ * @brief   Iterates over all forwarding table entries in the NIB
+ *
+ * @pre `(state != NULL) && (fte != NULL)`
+ *
+ * @param[in] next_hop      Restrict iteration to entries to this next hop. NULL
+ *                          for any next hop. Can be used to build a source
+ *                          routing tree.
+ * @param[in] iface         Restrict iteration to entries on this interface.
+ *                          0 for any interface.
+ * @param[in,out] state     Iteration state of the forwarding table. Must point
+ *                          to a NULL pointer to start iteration.
+ * @param[out] fte          The next forwarding table entry.
+ *
+ * The iteration over all forwarding table entries in the NIB includes all
+ * entries added via @p gnrc_ipv6_nib_ft_add() and entries that are currently
+ * in the Destination Cache, in the Prefix List, and in the Default Router List.
+ *
+ * Usage example:
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * #include "net/gnrc/ipv6/nib/ft.h"
+ *
+ * int main(void) {
+ *     void *state = NULL;
+ *     gnrc_ipv6_nib_ft_t fte;
+ *
+ *     puts("My neighbors:");
+ *     while (gnrc_ipv6_nib_ft_iter(NULL, 0, &state, &fte)) {
+ *         gnrc_ipv6_nib_ft_print(&fte);
+ *     }
+ *     return 0;
+ * }
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * @note    The list may change during iteration, but no duplication of already
```

Comment:
Are you sure the latter part of this sentence is true? What happens if an entry gets removed and then added (after 3 other additions). Then the same value would end up later in the list, right?

Reply:
Yes, that may happen. But then it is also wrong for `nib_pl_iter()` and `nib_nc_iter()`. Shall I remove it?

Reply:
>Shall I remove it?

Yup. No need to advertise features we cannot guarantee (: